src/async_fuse/memfs/s3_metadata.rs

```diff
@@ -186,11 +201,52 @@ impl<S: S3BackEnd + Sync + Send + 'static> MetaData for S3MetaData<S> {
     }
 
     #[instrument(skip(self))]
-    async fn flush(&self, ino: u64, fh: u64) -> DatenLordResult<()> {
+    async fn flush(&self, ino: u64, _fh: u64) -> DatenLordResult<()> {
+        let (local_mtime, local_size) = {
+            let guard = pin();
+
+            (
+                self.local_mtime.remove_with_guard(&ino, &guard).copied(),
+                self.local_size.remove_with_guard(&ino, &guard).copied(),
+            )
+        };
+
         retry_txn!(TXN_RETRY_LIMIT, {
             let mut txn = self.kv_engine.new_meta_txn().await;
             let mut inode = self.get_inode_from_txn(txn.as_mut(), ino).await?;
-            inode.flush(ino, fh).await;
+
+            if inode.is_deferred_deletion() {
```

Comment:
The flush should correspond to the close() operation, and I don't think it deletes the file. It looks like we need to investigate the maintenance of `open_count` and `lookup_count`.

Reply:
Yes, this is not to delete a file. But the deferred deletion is marked by `unlink()`, when `open_count` or `lookup_count` is **not** zero. Here checks the flag, and if it's true, there is no need to flush the file. This is following the old implementation of `flush()`.